libs/blocks/global-navigation/global-navigation.js

```diff
@@ -80,17 +80,32 @@ export const CONFIG = {
         name: 'profile',
         attributes: {
           isSignUpRequired: false,
+          messageEventListener: (event) => {
+            const { name, payload, executeDefaultAction } = event.detail;
```

Comment:
I would love to have some checks in place and make sure `executeDefaultAction` is actually a Promise and it's being sent as part of the event.detail, it doesn't hurt to be extra careful.